deps/v8/src/base/platform/platform-solaris.cc

```diff
@@ -70,7 +70,7 @@ Stack::StackSlot Stack::GetStackStart() {
   pthread_attr_t attr;
   int error;
   pthread_attr_init(&attr);
-  error = pthread_attr_get_np(pthread_self(), &attr);
+  error = 0;
```

Comment:
I think this change should be sent to v8 instead here:
https://chromium.googlesource.com/v8/v8.git/+/refs/heads/master/src/base/platform/platform-solaris.cc#73
as this is a part of a vendor dependency.

Reply:
OK, probably. I haven't studied it that much. So, how should we contact them? What should we tell them?

Reply:
You may follow the [V8 Contributing Guide](https://v8.dev/docs/contribute) and open a PR with this change. Once it gets landed in V8, we can update the code here with your commit and it will not get overwritten when a new V8 update PR lands.

Reply:
Based on some cursory searching, the correct Solaris way of getting the stack start is to skip all the pthread stuff, and just do:

```
stack_t s;
thr_stksegment(&s);
return s.ss_sp;
```

I don't have a Solaris machine to test this with though, so I can't vouch for it.

Reply:
> Based on some cursory searching, the correct Solaris way of getting the stack start is to skip all the pthread stuff, and just do:
> 
> ```
> stack_t s;
> thr_stksegment(&s);
> return s.ss_sp;
> ```
> 
> I don't have a Solaris machine to test this with though, so I can't vouch for it.

Well, I have a Solaris machine, can I somehow help? As I said, I am not a programmer, though.

Reply:
> You can replace the entire Stack::GetStackStart() method with the above, i.e. change it to
> 
> ```
> Stack::StackSlot Stack::GetStackStart() {
>   stack_t s;
>   thr_stksegment(&s);
>   return s.ss_sp;
> }
> ```
> 
> and report back whether it a) compiles, and b) works.

Ah, you think there isn't a simpler test? Compiling NodeJS takes about an hour.

Reply:
It's the best idea I've got, sorry.

Reply:
> It's the best idea I've got, sorry.

OK, I will try. By the way, do you think I can somehow speed up my Solaris Virtual Machine in VirtualBox? Do you think giving it more than 2GB of RAM (as it has now) is worth it? I know that, if I give a virtual machine 4GB of RAM, I sometimes get a Kernel Panic on the host, even though it has 8 GB of RAM. It is significantly slower in QEMU than in VirtualBox, for some reason.

Reply:
In that case try changing `#include <pthread.h>` at the top to `#include <thread.h>`; if that doesn't work then sorry, I'm out of ideas.

Reply:
> In that case try changing `#include <pthread.h>` at the top to `#include <thread.h>`; if that doesn't work then sorry, I'm out of ideas.

Excellent, now it compiles and works. That is, the [Analog Clock program I have made](https://github.com/FlatAssembler/AECforWebAssembly/blob/master/analogClock/analogClock.js) works, I don't know how to test if everything will work.